tests/functional/otlp-metrics/04-metrics-traces-gen.yaml

```diff
@@ -0,0 +1,81 @@
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: otelgen-traces
+spec:
+  completions: 1
+  parallelism: 1
+  template:
+    metadata:
+      labels:
+        app: otelgen
+    spec:
+      containers:
+        - name: otelgen-traces-multi
+          image: ghcr.io/krzko/otelgen:latest
```

Comment:
How about using the official upstream image?

Reply:
[Otelgen](https://github.com/krzko/otelgen) provides both OTLP metrics and traces generation, so kept the image  for both traces and metrics deployment and job. 

This test case depends on the OpenShift monitoring stack like using the user workload monitoring feature. So added e2e-openshift to the Makefile same as what we have done for the Tempo Operator to run the OpenShift specific cases. Also added the note in CONTRIBUTING.md file. This will help us to keep all our test cases in upstream.

Reply:
I had a question about the CHANGELOG.md. Do we add a changelog file for adding test cases or skip it ?